### DISCUSSION TITLE
Is this the worse software ever?

### DISCUSSION THREAD

**Discussion by btahir** [clerk/javascript#2076] [Support request - DO NOT OPEN HERE]
Seriously - 

What were you smoking when you decided to make every route private? Next.js Middleware is completely broken, huge performance hits. Vercel builds failing because of a 250MB limit being exceeded since 13.4.9 (past 3 months). Millions raised and no one gives a F***. What the hell are you guys doing?!
I want to leave you but much like someone stuck with an abusive spouse I am stuck with you for this project because the effort to switch is too much right now.

Please please please. All I want is a functioning app where /dashboard(.*) is protected and you leave everything else the f*** alone. Can you do this for Next.14?

**Comment by colinclerk** [comment]
Hey Bilal,

Sorry for the frustration.  I just put [an example repo up for your use-case here](https://github.com/colinclerk/protected_dashboard).

Here's a snippet you can use for your middleware:

```ts
import { authMiddleware, redirectToSignIn } from "@clerk/nextjs";

export default authMiddleware({
  afterAuth(auth, req, evt) {
    // handle users who aren't authenticated and access /dashboard
    if (!auth.userId && req.nextUrl.pathname.startsWith("/dashboard")) {
      return redirectToSignIn({ returnBackUrl: req.url });
    }
  },
});

export const config = {
  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
};
```

Let me know if this works, and I can help with anything else!

**Reply by btahir** [reply]
It does not. Or it does if you want to never upgrade from Next 13.4.9. I can't give you a toy example because I think what breaks it is that I have a lot of blog pages. Trying different variations of excluding it in matcher, add in ignoredPaths only works until that version. For all the later versions of Next Vercel will not build because of a hitting the limit of a function exceeding 250MB in size.

**Reply by colinclerk** [reply]
The example I put together is Next 14, and I used an `afterAuth` rather than ignored routes.  Are you using any particular Next features that might be causing problems?  Happy to poke into a repro.

**Reply by btahir** [reply]
All I know is this works up till 13.4.9
```
import { authMiddleware } from '@clerk/nextjs'

export default authMiddleware({
  publicRoutes: ['/', '/api/checkout', '/sign-in', '/sign-up', '/pricing', '/faq'],
  ignoredRoutes: ['/blog(.*)']
})

export const config = {
  matcher: ['/((?!.+\\.[\\w]+$|_next|/blog).*)', '/api/checkout', '/dashboard(.*)', '/pricing'],
}
```

and if I try to upgrade Next it breaks.

**Reply by colinclerk** [reply]
That's bizarre. What version of `@clerk/nextjs`?  If it's not `4.26.2`, can you try upgrading?  We did have some issues with 14 when released and have since republished to resolve.

If the error persists after upgrade, can you please share the specific error here? I'll ask the team to take a look.

In the interim though, if it's just a middleware error, the above snippet can potentially get you through on 14.

**Reply by btahir** [reply]
Man I've upgraded to 4.26.2 and is still happening and has been happening for the past 3 months (not just with 14 but every version since the one I said). Yeah this gets me through but my app is now stuck on this version of Next because of this and has been for the past few months. Obviously not sustainable.

This is the error:
```
Error: A Serverless Function has exceeded the unzipped maximum size of 250 MB. : https://vercel.link/serverless-function-size
```

Thanks and hope this gets resolved finally.

**Reply by colinclerk** [reply]
Hey Bilal - the error you shared relates to the Serverless Functions, while middleware runs on Vercel's Edge Runtime.  That indicates the error is coming from elsewhere besides the middleware.

Our instinct is this doesn't relate to Clerk, but likely a change in how Next.js handles tree shaking between those versions. Is there anything in particular leading you to believe it's an issue with Clerk middleware?  Two things that would help us diagnose this further are:
1. A more complete output from your console showing the error that was triggered
2. A complete list of the imports in the problematic file.  (If our tree-shaking idea is correct, our experience is that it's generally because an import is inadvertently bundled with a Serverless Function)
